backend/database-router.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const readConfig = () => {
+  const configString = process.env.GI_PERSIST_ROUTER_CONFIG
+  const config = JSON.parse(configString)
+
+  if (!config['*']) {
+    throw new Error('Fallback storage (*) is required')
+  }
+
+  return config
+}
+let config
+const backends = Object.create(null)
+
+const lookupBackend = (key) => {
+  const keys = Object.keys(config)
+  for (let i = 0; i < keys.length; i++) {
+    if (key.slice(0, keys[i].length) === keys[i]) {
```

Comment:
Why are we slicing the key here, instead of doing `key.startsWith(keys[i])` ? Even then, I fail to understand why a plain `key === keys[i]` isn't enough.

Reply:
1. `key.startsWith(keys[i])` would also work
2. `key === keys[i]` isn't enough because the key in config is a prefix. Actually, maybe `keyPrefixes` and `keyPrefix` would be better names.

Reply:
> key.startsWith(keys[i]) would also work

Can you check which is faster?

Reply:
I think either is about as fast. Perhaps `startsWith` has the potential to be slightly faster (as the equivalent slicing would be done by native code) and is perhaps more readable.